sql/executor.go

```diff
@@ -39,11 +42,18 @@ var errTransactionInProgress = errors.New("there is already a transaction in pro
 type Executor struct {
 	db     client.DB
 	nodeID uint32
+
+	// System Config and mutex.
+	systemConfig   *config.SystemConfig
```

Comment:
In this case there's not much difference between a mutex and an atomic.Value (since the gossip process means that the system config is not strongly synchronized with anything else), but in general I prefer a mutex as the default synchronization mechanism because they give you more options for composition: you could add more fields to be protected by the same mutex, or block all readers during an update process.